addons/website_portal_sale/models/account_invoice.py

```diff
@@ -1,12 +1,24 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, exceptions, models
+from odoo import api, exceptions, fields, models
 
 
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
+    mail_url = fields.Char(
+        'Website URL', compute='_compute_mail_url',
+        help='The full URL to access the document through the website.')
+
+    def _compute_mail_url(self):
+        for inv in self:
+            url = '/mail/view?model=%s&res_id=%s' % (self._name, inv.id)
+            auth_param = inv.partner_id.url_get_user_login()
+            if auth_param:
+                url = url + '&' + auth_param[inv.partner_id.id]
```

Comment:
Not sure I catch your drift.

Would you like me to create a {model: self._name, res_id, inv.id} and url_encode that ?

Or do you want to do the encoding of url_get_user_login() here and not in its method ?